testing/test.js

```diff
@@ -0,0 +1,105 @@
+  //duration column
+  var duration = head_row.insertCell(2);
+  duration.className = 'sortable numeric asc inactive';
+  duration.setAttribute('col', 'duration');
+  duration.innerHTML = '<div class="sort-icon">vvv</div> Duration';
+  //Links
+  var links = head_row.insertCell(3);
+  links.innerHTML = "Links";
+
+  //Failed row
+  var tbody_failed = document.createElement('tbody');
+  tbody_failed.className = 'failed results-table-row';
+  //columns
+  var failed_row = tbody_failed.insertRow(0);
+  var col_result_failed = failed_row.insertCell(0);
+  col_result_failed.className = 'col-result';
+  col_result_failed.innerHTML = 'Failed';
+  var col_name_failed = failed_row.insertCell(1);
+  col_name_failed.className = 'col-name';
+  col_name_failed.innerHTML = 'test_documentation.py::test_fail';
+  var col_duration_failed = failed_row.insertCell(2);
+  col_duration_failed.className = 'col-duration';
+  col_duration_failed.innerHTML = '0.00';
+  var col_links_failed = failed_row.insertCell(3);
+  col_links_failed.className = 'col-links';
+  // extra and log
+  var row_extra_failed = tbody_failed.insertRow(0);
+  var col_extra_failed = row_extra_failed.insertCell(0);
+  col_extra_failed.className = 'extra';
+  col_extra_failed.innerHTML = '<div class="log"></div>';
+
+  table.appendChild(thead);
+  table.appendChild(tbody_failed);
+
+  var html = document.createElement('html');
+  var body = document.createElement('body');
+  body.appendChild(table);
+  html.appendChild(body);
+  return html;
+}
+
+QUnit.test( 'toArray', function(assert) {
+  var nodeList = new NodeList();
+  assert.ok(toArray(nodeList) instanceof Array);
+  assert.ok(toArray(null) === null);
+});
+
+QUnit.test('find', function (assert) {
+  var table = create_elem_table();
+  assert.ok(find('thead', table).innerText ===
+            'vvv Resultvvv Testvvv DurationLinks');
+  assert.ok(find('.sort-icon', table).innerHTML === 'vvv');
+  assert.ok(find('#results-table-head', table) != null);
+  assert.ok(find('#results-table', table) != null);
+  assert.ok(find('.not-in-table', table) === null);
+});
+
+QUnit.test('find_all', function(assert) {
+  var table = create_elem_table();
+  assert.ok(find_all('.sort-icon', table).length === 3);
+  assert.ok(find_all('.not-in-table', table).length === 0);
+});
+
+QUnit.test('toggle_sort_states', function(assert){
+  var table = create_elem_table();
+  var row_numeric = find('.numeric', table);
+  assert.ok(row_numeric.className === 'sortable numeric asc inactive');
+  toggle_sort_states(row_numeric);
+  assert.ok(row_numeric.className === 'sortable numeric asc active');
+  toggle_sort_states(row_numeric);
+  assert.ok(row_numeric.className === 'sortable numeric active desc');
+});
+
+QUnit.test('sort_column_numeric', function(assert){
+  var table = create_elem_table();
+  setDocument(table);
+  row_numeric = find('.numeric', table);
+  sort_column(row_numeric);
+  assert.ok(0 == 0);
```

Comment:
This test is not finished yet, I am waiting for your comments if you think this is a good practice.